cmd/internal/cli/inspect_linux.go

```diff
@@ -264,132 +324,259 @@ var InspectCmd = &cobra.Command{
 			sylog.Fatalf("container not found: %s", err)
 		}
 
-		abspath, err := filepath.Abs(args[0])
+		if !runscript && !helpfile && !deffile && !testfile && !environment {
+			labels = true
+		}
+
+		fimg, err := sif.LoadContainer(args[0], true)
 		if err != nil {
-			sylog.Fatalf("While determining absolute file path: %v", err)
+			sylog.Fatalf("failed to load SIF container file: %s", err)
 		}
-		name := filepath.Base(abspath)
+		defer fimg.UnloadContainer()
 
-		a := []string{"/bin/sh", "-c", ""}
+		var inspectData string
+		inspectDataJSON := make(map[string]map[string]string, 1)
 
-		if listApps {
-			sylog.Debugf("Listing all apps in container")
-			a[2] += listAppsCommand
+		if deffile || runscript || helpfile || environment {
+			inspectDataJSON["container"] = make(map[string]string, 1)
 		}
 
-		// If AppName is given fail quickly (exit) if it doesn't exist
-		if AppName != "" {
-			sylog.Debugf("Inspection of App %s Selected.", AppName)
-			a[2] += getAppCheck(AppName)
+		// Inspect Labels
+		if labels {
+			sifData, _, err := getMetaData(&fimg, sif.DataLabels)
+			if err == ErrNoMetaData {
+				sylog.Warningf("No metadata partition, searching in container...")
+				inspectInContainer(args[0])
+			} else if err != nil {
+				sylog.Fatalf("Unable to get label metadata: %s", err)
+			}
+			inspectDataJSON["labels"] = make(map[string]string, 1)
+
+			for _, v := range sifData {
+				metaData := v.GetData(&fimg)
+				var hrOut map[string]*json.RawMessage
+				err := json.Unmarshal(metaData, &hrOut)
+				if err != nil {
+					sylog.Fatalf("Unable to get json: %s", err)
+				}
+				inspectData += "== labels ==\n"
+				for k := range hrOut {
+					inspectData += fmt.Sprintf("%s: %s\n", k, string(*hrOut[k]))
+					inspectDataJSON["labels"][k] = string(*hrOut[k])
+				}
+			}
 		}
 
-		if helpfile {
-			sylog.Debugf("Inspection of helpfile selected.")
-			a[2] += getHelpCommand(AppName)
+		// Inspect Deffile
+		if deffile {
+			sifData, _, err := getMetaData(&fimg, sif.DataDeffile)
+			if err != nil {
+				sylog.Fatalf("Unable to get metadata: %s", err)
+			}
+			for _, v := range sifData {
+				metaData := v.GetData(&fimg)
+				data := string(metaData)
+				inspectDataJSON["container"]["deffile"] = data
+				inspectData += data
+			}
 		}
 
-		if deffile {
-			sylog.Debugf("Inspection of deffile selected.")
-			a[2] += getDefinitionCommand()
+		// For %help
+		if helpfile {
+			sifData, _, err := getMetaData(&fimg, sif.DataDeffile)
+			if err != nil {
+				sylog.Fatalf("Unable to get metadata: %s", err)
+			}
+
+			for _, v := range sifData {
+				metaData := v.GetData(&fimg)
+				data := parseDeffile(string(metaData), "%help")
+				inspectDataJSON["container"]["helpfile"] = data
+				inspectData += "== helpfile ==\n" + data
+			}
 		}
 
+		// For %runscript
 		if runscript {
```

Comment:
I think its okay if we just list the runscript from the deffile, we can add/change the help for `inspect` to something like: 

```
--runscript    show the runscript from the deffile
```

Also, I'm trying to avoid running/shelling into the container for inspect, so we can inspect a container on a different architecture, or a encrypted container.